editoast/src/models/round_trips.rs

```diff
@@ -0,0 +1,87 @@
+use diesel::QueryDsl;
+use diesel_async::RunQueryDsl;
+use editoast_derive::Model;
+use futures_util::stream::TryStreamExt;
+use std::ops::DerefMut;
+
+#[derive(Clone, Debug, Model)]
+#[model(table = database::tables::train_schedule_round_trips)]
+#[model(gen(batch_ops = c, list))]
+pub struct TrainScheduleRoundTrips {
+    pub id: i64,
+    /// First ID of the train schedule of this round trip
+    pub left_id: i64,
+    /// Second ID of the train schedule of this round trip
+    /// This is `None` for one-way trains
+    pub right_id: Option<i64>,
+}
+
+impl DeleteBatch<i64> for TrainScheduleRoundTrips {
+    type Error = editoast_models::model::Error;
+
+    /// Deletes a batch of train schedule round trips given a list of train schedule IDs
+    async fn delete_batch<I: IntoIterator<Item = i64> + Send>(
+        conn: &mut DbConnection,
+        train_schedule_ids: I,
+    ) -> Result<usize, Self::Error> {
+        use database::tables::train_schedule_round_trips::dsl;
+        use diesel::prelude::*;
+        use diesel_async::RunQueryDsl;
+        use std::ops::DerefMut;
+
+        let ids = train_schedule_ids.into_iter().collect::<Vec<_>>();
+        diesel::delete(
+            database::tables::train_schedule_round_trips::table
+                .filter(dsl::left_id.eq_any(&ids).or(dsl::right_id.eq_any(&ids))),
+        )
+        .execute(conn.write().await.deref_mut())
+        .await
+        .map_err(editoast_models::model::Error::from)
+    }
+}
+
+impl TrainScheduleRoundTrips {
+    #[tracing::instrument(
+        name = "list_paginated<TrainScheduleRoundTrips>",
+        skip_all,
+        err,
+        fields(timetable_id, limit, offset)
+    )]
+    pub async fn list_paginated(
+        conn: &mut DbConnection,
+        timetable_id: i64,
+        page: u64,
+        page_size: u64,
+    ) -> crate::error::Result<(Vec<Self>, PaginationStats)> {
+        use database::tables::train_schedule;
+        use database::tables::train_schedule_round_trips;
+
+        let count = train_schedule_round_trips::table
+            .inner_join(train_schedule::table)
+            .filter(train_schedule::dsl::timetable_id.eq(timetable_id))
+            .count()
+            .get_result::<i64>(conn.write().await.deref_mut())
+            .await
+            .map_err(editoast_models::model::Error::from)?;
+
+        let offset = (page - 1) * page_size;
+        let results: Vec<TrainScheduleRoundTrips> = train_schedule_round_trips::table
+            .inner_join(train_schedule::table)
+            .filter(train_schedule::dsl::timetable_id.eq(timetable_id))
+            .select(train_schedule_round_trips::all_columns)
+            .limit(page_size as i64)
+            .offset(offset as i64)
+            .load_stream::<TrainScheduleRoundTripsRow>(conn.write().await.deref_mut())
+            .await?
+            .map_ok(<TrainScheduleRoundTrips as crate::models::prelude::Model>::from_row)
+            .try_collect()
+            .await?;
```

Comment:
You can use `models::pagination::load_for_pagination` to simplify things. I forgot it existed when we discussed the issue earlier, sorry 😅

(If you decide to use it could you instrument the function? 👀)